src/player_activity.cpp

```diff
@@ -464,3 +466,28 @@ void player_activity::inherit_distractions( const player_activity &other )
         ignore_distraction( type );
     }
 }
+
+
+std::map<distraction_type, std::string> player_activity::get_distractions()
```

Comment:
This function is a good candidate for a simpler and more focused unit test, now that you've factored it out of `do_turn`.

It's great to have the sort of "end-to-end" (which you've already done) that sets up the activity and actually confirms that interruption works during activity processing, but because this function is now isolated from the activity processing stuff, it might be easier to test it directly, checking that it returns the expected distraction type and message in different field/monster situations.

Reply:
Yeah, classic case of not seeing the forest for the trees.

There's still worth in checking this for different activities, but no actual processing is needed. Simply constructing and starting the activity is enough. I guess the real work here will be making a list with all/many of the activities. Then just run some standard cases with and without monsters and fields on them.